Content.Server/SqueezeBottle/SqueezeBottleSystem.cs

```diff
@@ -0,0 +1,93 @@
+using Content.Shared.Database;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Item;
+using Content.Shared.SqueezeBottle;
+using Content.Shared.Nutrition.EntitySystems;
+using Content.Shared.Popups;
+using Content.Shared.Verbs;
+using Robust.Shared.Audio.Systems;
+
+using Content.Server.ReagentOnItem;
+
+namespace Content.Server.SqueezeBottle;
+
+public sealed class SqueezeBottleSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly OpenableSystem _openable = default!;
+    [Dependency] private readonly ReagentOnItemSystem _reagentOnItem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SqueezeBottleComponent, AfterInteractEvent>(OnInteract, after: new[] { typeof(OpenableSystem) });
+        SubscribeLocalEvent<SqueezeBottleComponent, GetVerbsEvent<UtilityVerb>>(OnUtilityVerb);
+    }
+
+    private void OnInteract(Entity<SqueezeBottleComponent> entity, ref AfterInteractEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!args.CanReach || args.Target is not { Valid: true } target)
+            return;
+
+        if (TryToApplyReagent(entity, target, args.User))
+            args.Handled = true;
+    }
+
+    private void OnUtilityVerb(Entity<SqueezeBottleComponent> entity, ref GetVerbsEvent<UtilityVerb> args)
+    {
+        if (!args.CanInteract || !args.CanAccess || args.Target is not { Valid: true } target ||
+        _openable.IsClosed(entity))
+            return;
+
+        var user = args.User;
+
+        var verb = new UtilityVerb()
+        {
+            Act = () => TryToApplyReagent(entity, target, user),
+            IconEntity = GetNetEntity(entity),
+            Text = Loc.GetString("squeeze-bottle-verb-text"),
+            Message = Loc.GetString("squeeze-bottle-verb-message")
+        };
+
+        args.Verbs.Add(verb);
+    }
+
+    // Try to apply the reagent thats stored inside the squeeze bottle into an object.
+    private bool TryToApplyReagent(Entity<SqueezeBottleComponent> entity, EntityUid target, EntityUid actor)
+    {
+
+        if (!HasComp<ItemComponent>(target))
+        {
+            _popup.PopupEntity(Loc.GetString("squeeze-bottle-not-item-failure", ("target", Identity.Entity(target, EntityManager))), actor, actor, PopupType.Medium);
+            return false;
+        }
+
+        if (HasComp<ItemComponent>(target) && _solutionContainer.TryGetSolution(entity.Owner, entity.Comp.Solution, out var solComp, out var solution))
+        {
+            var reagent = solution.SplitSolution(entity.Comp.AmountConsumedOnUse);
```

Comment:
Here's the source of the problem with the visuals not updating.

Because solutions can exist outside of an entity or component, splitting a solution with Solution.SplitSolution doesn't inform the entity about the change.

Simply split the solution with the SharedSolutionContainerSystem.SplitSolution method instead - it calls UpdateAppearance, raises a SolutionChangedEvent on the entity, and all that.

```suggestion
            var reagent = _solutionContainer.SplitSolution(solComp.Value, entity.Comp.AmountConsumedOnUse);
```

Reply:
Thank you so much! I wish I had asked earlier it has been annoying me for the last two weeks 😭